packages/apm/src/integrations/tracing.ts

```diff
@@ -14,6 +14,73 @@ import { Span as SpanClass } from '../span';
 import { SpanStatus } from '../spanstatus';
 import { Transaction } from '../transaction';
 
+/** Holds the latest LargestContentfulPaint value (it changes during page load). */
+let lcp: { [key: string]: any };
+
+/** Force any pending LargestContentfulPaint records to be dispatched. */
+let forceLCP = () => {
+  /* No-op, replaced later if LCP API is available. */
+};
```

Comment:
I decided to put this on the top-level for now instead of inside of Tracing because:

1. We want `firstHiddenTime` to be setup as soon as possible.
2. Most of the LCP code can live in one place (we can move it somewhere else) instead of spread across `Tracing` (static fields, static methods, etc)
3. Less indentation


Everything else is inside of a block, so out of the global namespace.

- `lcp` is updated async via `PerformanceObserver` when the relevant API is supported by the browser (limited support).
- `forceLCP` is called right before reading `lcp`, as per the reference implementation.

Reply:
> We want firstHiddenTime to be setup as soon as possible. 
While technically correct everything here is synchronous and also everything else is inside the class, I would prefer having this also in there.

Reply:
Moved.